tests/golomb_ruler.cpp

```diff
@@ -0,0 +1,116 @@
+/* Copyright 2017-2018 PaGMO development team
+
+This file is part of the PaGMO library.
+
+The PaGMO library is free software; you can redistribute it and/or modify
+it under the terms of either:
+
+  * the GNU Lesser General Public License as published by the Free
+    Software Foundation; either version 3 of the License, or (at your
+    option) any later version.
+
+or
+
+  * the GNU General Public License as published by the Free Software
+    Foundation; either version 3 of the License, or (at your option) any
+    later version.
+
+or both in parallel, as here.
+
+The PaGMO library is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received copies of the GNU General Public License and the
+GNU Lesser General Public License along with the PaGMO library.  If not,
+see https://www.gnu.org/licenses/. */
+
+#define BOOST_TEST_MODULE golomb_ruler_test
+#include <boost/test/included/unit_test.hpp>
+
+#include <boost/lexical_cast.hpp>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+
+#include <pagmo/algorithm.hpp>
+#include <pagmo/population.hpp>
+#include <pagmo/problem.hpp>
+#include <pagmo/algorithms/sga.hpp>
+#include <pagmo/problems/golomb_ruler.hpp>
+#include <pagmo/problems/unconstrain.hpp>
+#include <pagmo/types.hpp>
+
+using namespace pagmo;
+
+BOOST_AUTO_TEST_CASE(golomb_ruler_test)
+{
+    // Problem construction
+    BOOST_CHECK_THROW((golomb_ruler{0u, 10u}), std::invalid_argument);
+    BOOST_CHECK_THROW((golomb_ruler{10u, 0u}), std::invalid_argument);
+    BOOST_CHECK_THROW((golomb_ruler{2300u, 1421552366u}), std::overflow_error);
```

Comment:
Would be probably best to use ``std::numeric_limits`` here to trigger the error, in order to make the test portable.